backend/src/backend/api/file_routes.py

```
@@ -0,0 +1,68 @@
+
+        # Save file metadata to database
+        uploaded_file = save_file(file_id, file.filename, file_type_enum, session)
+        logger.info(f"Successfully uploaded file {file_id}")
+        return uploaded_file
+        
+    except Exception as e:
+        # Clean up file if saved
+        if "file_path" in locals() and os.path.exists(file_path):
+            os.remove(file_path)
+        log_traceback(e, f"upload_file endpoint - filename={file.filename}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)